toogoodtogo_ha_mqtt_bridge/main.py

```diff
@@ -395,6 +401,103 @@ def calc_timeout():
         exit_from_thread("Invalid cron schedule", 1)
 
 
+def intense_fetch():
+    if (
+        "intense_fetch" not in settings.tgtg
+        or "period_of_time" not in settings.tgtg.intense_fetch
+        or "interval" not in settings.tgtg.intense_fetch
+    ):
+        logger.error("Incomplete settings file. Please check the sample!")
+        return None
+
+    if settings.tgtg.intense_fetch.period_of_time > 60:
+        logger.warning(
+            "Stopped intense fetch. Maximal intense fetch period time are 60 minutes. Reduce your setting!"
+        )
+        return None
+
+    if settings.tgtg.intense_fetch.interval < 10:
+        logger.warning(
+            "Stopped intense fetch. Minimal intense fetch interval are 10 seconds. Increase your setting!"
+        )
+        return None
+
+    mqtt_client.publish(
+        f"homeassistant/switch/toogoodtogo_intense_fetch/state",
+        "ON",
+    )
+
+    t = threading.currentThread()
+    t_end = time.time() + 60 * settings.tgtg.intense_fetch.period_of_time
+
+    while time.time() < t_end and getattr(t, "do_run", True):
+        logger.info("Intense fetch started")
+        if not check():
+            logger.error("Intense fetch was not successfully")
+        else:
+            logger.info("Intense fetch finished")
+            sleep(settings.tgtg.intense_fetch.interval)
+
+    global intense_fetch_thread
+    intense_fetch_thread = None
+
+    mqtt_client.publish(
+        f"homeassistant/switch/toogoodtogo_intense_fetch/state",
+        "OFF",
+    )
+
+    logger.info("Intense fetch stopped")
+
+
+def on_message(client, userdata, message):
+    global intense_fetch_thread
+    if message.topic.endswith("toogoodtogo_intense_fetch/set"):
+        if message.payload.decode("utf-8") == "ON":
+            if intense_fetch_thread:
+                logger.error("Intense fetch thread already running. Doing nothing.")
+                return None
+
+            thread = threading.Thread(target=intense_fetch)
+            intense_fetch_thread = thread
+            thread.start()
+        elif message.payload.decode("utf-8") == "OFF":
+            if intense_fetch_thread:
+                intense_fetch_thread.do_run = False
+                logger.info("Intense fetch is stopped in the next cycle.")
+                mqtt_client.publish(
+                    f"homeassistant/switch/toogoodtogo_intense_fetch/state",
+                    "OFF",
+                )
+            else:
+                logger.info("No running thread found. Doing nothing.")
+
+
+def register_fetch_sensor():
+    mqtt_client.publish(
+        f"homeassistant/switch/toogoodtogo_bridge/intense_fetch/config",
+        json.dumps(
+            {
+                "name": "TooGoodToGo - Intense fetch",
```

Comment:
I would propose to use just `Intense fetch` as name, it seeems to be trunkated

![image](https://user-images.githubusercontent.com/5927148/174662925-4532d287-87f4-4785-a66d-883d028467b0.png)

Reply:
Done with https://github.com/MaxWinterstein/toogoodtogo-ha-mqtt-bridge/pull/60/commits/4028f1fc4b968e439bb546cd541cf4dea18ae084